src/mbgl/renderer/render_layer.hpp

```diff
@@ -28,20 +28,24 @@ class LayerRenderData {
     Immutable<style::LayerProperties> layerProperties;
 };
 
-class LayerPlacementData {
+class SortKeyRange {
+public:
+    bool isFirstRange() const { return start == 0u; }
+    float sortKey;
+    size_t start;
+    size_t end;
+};
+
+class BucketPlacementData {
 public:
-    friend bool operator<(const LayerPlacementData& lhs, const LayerPlacementData& rhs) {
-        return lhs.sortKey < rhs.sortKey;
-    }
     std::reference_wrapper<Bucket> bucket;
     std::reference_wrapper<const RenderTile> tile;
     std::shared_ptr<FeatureIndex> featureIndex;
-    bool firstInBucket;
-    float sortKey;
-    size_t symbolInstanceStart;
-    size_t symbolInstanceEnd;
+    optional<SortKeyRange> sortKeyRange;
 };
 
+using LayerPlacementData = std::list<BucketPlacementData>;
```

Comment:
In a last PR, std::vector was switched to std::list, could we use vector instead?

Reply:
discussed offline, needs to be stable.